src/Massive.Shared.cs

```diff
@@ -29,11 +29,15 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Specialized;
+#if !COREFX
```

Comment:
It's better to use
```
#if !NETSTANDARD1_6
// code that has to run on .net full
#endif
```
as that default compilation directive is defined in the .net core / netstandard projects. It should really target netstandard1.6, not 'net core' as that's a better directive: netstandard1.6 is supported by .net core, but also on full (4.6.2) and also on UWP (if I'm not mistaken). Converting it to .net core 2.0 is then easier: the focus is then netstandard2.0, not 1.6.